File: practice-jwt-back/src/user/user.entity.ts
import { Entity, PrimaryGeneratedColumn, Column, CreateDateColumn, UpdateDateColumn } from 'typeorm';
import {IsString} from "class-validator";

@Entity('users')
export class UserEntity {
  @PrimaryGeneratedColumn('uuid')
  id: string;


  @Column({ unique: true })
  @IsString({
    message: 'email은 String 타입을 입력해줘야 합니다.'
  })
  email: string;

  @Column()
  @IsString({
    message: 'password는 String 타입을 입력해줘야 합니다.'
  })
  password: string;

  @CreateDateColumn()
  createdAt: Date;

  @UpdateDateColumn()
  updatedAt: Date;
}

File: practice-jwt-front-flutter/lib/domain/entities/user_entity.dart
import 'package:freezed_annotation/freezed_annotation.dart';

part 'user_entity.freezed.dart';

@freezed
class UserEntity with _$UserEntity {
  const factory UserEntity({
    required String uuid,
    required String email,
    required String password,
  }) = _UserEntity;
}
